cst438_registration/src/test/java/com/cst438/JunitStudentTest.java

```diff
@@ -0,0 +1,301 @@
+				continue;
+			}
+		}
+	scan.close();
+	}
+	
+	@Test
+	public void deleteStudentTest() throws Exception {
+		Scanner scan = new Scanner(System.in);
+		MockHttpServletResponse response;
+		List<Student> students = studentRepository.findAll();
+		List <Enrollment> enrolls = (List<Enrollment>) enrollmentRepository.findAll();
+		/*Student studentToDelete = new Student();
+		studentToDelete.setName("DeleteName");
+		studentToDelete.setStatusCode(1);
+		studentToDelete.setEmail("delete@email.com");
+		studentToDelete.setStatus("Testing for deleteStudent()");
+		String studentJsonTest = asJsonString(studentToDelete); 
+		
+		response = mvc.perform(
+				MockMvcRequestBuilders
+				.post("/addStudent")
+				.contentType(MediaType.APPLICATION_JSON)
+				.content(studentJsonTest)
+				.accept(MediaType.APPLICATION_JSON))
+				.andReturn().getResponse();
+		Student resultTest = fromJsonString(response.getContentAsString(),Student.class);
+		int testID = resultTest.getStudent_id();
+		System.out.println("IDS to simulate: " + testID);*/
+		int exit = 0;
+		
+		while (exit != 1) {
+			String ok = "y";
+			int id =0;
+			
+			do {
+				for(int i = 0;i<students.size();i++) {
+					System.out.println("ID:" +  students.get(i).getStudent_id());
+				}
+			System.out.print("\nEnter the id of the student to delete:");
+			id = scan.nextInt();
+			for (int i =0;i<enrolls.size();i++) {
+				if (enrolls.get(i).getStudent().getStudent_id() == id) {
+					if (enrolls.get(i).getStudent().getStatusCode() == 1) {
+						System.out.print("WARNING: This student is enrolled, are you sure you want to delete this student?(y/n)");
+						ok = scan.next();
+						if (ok == "n") {
+							System.out.print(String.format("Enter %d if you wish to delete another student, or %d if you wish to exit: ",0,1) );
+							exit = scan.nextInt();
+							if (exit == 1) {return;}
+						}
+					}
+				}
+			}
+			}while(ok != "y");
+			
+				response = mvc.perform(
+					MockMvcRequestBuilders
+					.delete("/deleteStudent/" + id)
+					.contentType(MediaType.APPLICATION_JSON)
+					.accept(MediaType.APPLICATION_JSON))
+					.andReturn().getResponse();
+				
+			if (response.getStatus() == 200) {
+				Optional<Student> student = studentRepository.findById(id);
+				assert(student.isEmpty());
+				System.out.print(String.format("Enter %d if you wish to delete another student, or %d if you wish to exit: ",0,1) );
+				exit = scan.nextInt();
+			}
+			else {
+				System.out.println("ERROR: Student not found in repository");
+				continue;
+			}
+		
+		}
+		scan.close();
+	}
+	
+private static <T> T  fromJsonString(String str, Class<T> valueType ) {
+	try {
+		return new ObjectMapper().readValue(str, valueType);
+	} catch (Exception e) {
+		throw new RuntimeException(e);
+	}
+}
+private static String asJsonString(final Object obj) {
+	try {
+		return new ObjectMapper().writeValueAsString(obj);
+	} catch (Exception e) {
+		throw new RuntimeException(e);
+	}
+	}
```

Comment:
The code includes integration tests for a Spring Boot application, covers various CRUD operations on student data, and uses Spring MockMvc for testing. It also demonstrates knowledge of JSON serialization and deserialization.